### PR TITLE
Allow default timezone for datetime input formats

### LINKED ISSUE #3171: Support specifying a timezone manually with strptime parser
For instance:
```yaml
name: timestamp
type: datetime
description: Time at which the event was emitted
input_formats:
  - rfc3339
  - unix_timestamp
  - "%Y %m %d %H:%M:%S.%f %z"
  - strptime:
    format: "%Y %m %d %H:%M:%S.%f"
    timezone: America/New_York
```

### PR DESCRIPTION
**Pull request by rdettai**
### Description

Closes #3171

### How was this PR tested?

Describe how you tested this PR.

### CODE REVIEW

File: quickwit/quickwit-datetime/src/date_time_format.rs [hunk 1]
```diff
@@ -208,18 +268,26 @@ impl FromStr for DateTimeInputFormat {
 impl Serialize for DateTimeInputFormat {
     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
     where S: serde::Serializer {
+        // if let DateTimeInputFormat::Strptime(parser) = self {
```
Comment (fulmicoton) [inline quickwit/quickwit-datetime/src/date_time_format.rs:271]: ?

File: quickwit/quickwit-datetime/src/date_time_format.rs [hunk 2]
```diff
@@ -176,11 +236,11 @@ impl DateTimeInputFormat {
     }
 }
 
-impl Display for DateTimeInputFormat {
-    fn fmt(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
-        formatter.write_str(self.as_str())
-    }
-}
+// impl Display for DateTimeInputFormat {
+//     fn fmt(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
```
Comment (fulmicoton) [inline quickwit/quickwit-datetime/src/date_time_format.rs:240]: ?

File: quickwit/quickwit-datetime/src/date_time_format.rs [hunk 3]
```diff
@@ -153,13 +177,49 @@ fn is_strftime_formatting(format_str: &str) -> bool {
...
+    String(String),
+    WithDefaultTimezone { format: String, default_tz: String },
+}
+
+impl TryFrom<StrptimeForSerde> for StrptimeParser {
+    type Error = String;
+
+    fn try_from(strptime_for_serde: StrptimeForSerde) -> Result<Self, Self::Error> {
+        match strptime_for_serde {
+            StrptimeForSerde::String(format) => StrptimeParser::from_str(&format),
```
Comment (fulmicoton) [inline quickwit/quickwit-datetime/src/date_time_format.rs:191]: serialization unit tests missing.

File: quickwit/quickwit-datetime/src/date_time_format.rs [hunk 4]
```diff
@@ -94,12 +112,18 @@ impl StrptimeParser {
     }
 
     pub fn parse_date_time(&self, date_time_str: &str) -> Result<OffsetDateTime, String> {
-        if *self.borrow_with_timezone() {
-            OffsetDateTime::parse(date_time_str, self.borrow_items()).map_err(|err| err.to_string())
+        if let TimezoneSource::Default(default_tz) = self.borrow_timezone_source() {
+            let primitive_datetime = self
+                .parse_primitive_date_time(date_time_str)
```
Comment (fulmicoton) [inline quickwit/quickwit-datetime/src/date_time_format.rs:117]: We can drop the self referencing bullshit by using the OwnedFormatItem of time rs

File: quickwit/quickwit-datetime/src/date_time_format.rs [hunk 5]
```diff
@@ -94,12 +112,18 @@ impl StrptimeParser {
...
+                .map_err(|err| err.to_string())?;
+            let parsed_time = primitive_datetime.assume_timezone(*default_tz);
+            if let OffsetResult::Some(datetime) = parsed_time {
```
Comment (fulmicoton) [inline quickwit/quickwit-datetime/src/date_time_format.rs:120]: This is missing a unit test.

**Comment by rdettai** [comment]
closing this for now until https://github.com/quickwit-oss/quickwit/issues/3171 is requested again

**Pull request closed without merging**